agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/ProfilingInitializer.java

```diff
@@ -46,46 +38,74 @@
  * </ul>
  */
 public class ProfilingInitializer {
+  private final AtomicBoolean currentlyEnabled = new AtomicBoolean();
+  private final String processId;
+  private final String machineName;
+  private final String roleName;
+  private final TelemetryClient telemetryClient;
+  private final String userAgent;
+  private final Configuration configuration;
+  private final File tempDir;
+
+  //////////////////////////////////////////////////////////
+  // These remain permanently live even if the profiler is disabled within the UI
```

Comment:
Is that a memory impact concern?

Reply:
Unfortunately there is no way around it, we need to constantly check the configuration to see if the user has enabled the profiler, so we need the infrastructure to do that polling